x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_details/forecasts_table/forecasts_table.js

```diff
@@ -163,6 +203,35 @@ export class ForecastsTable extends Component {
     await navigateToUrl(singleMetricViewerForecastLink);
   }
 
+  async deleteForecast(forecastId) {
+    const {
+      services: {
+        mlServices: { mlApi },
+      },
+    } = this.context;
+
+    this.setState({
+      isLoading: true,
+      isConfirmModalVisible: false,
```

Comment:
`forecastIdToDelete` needs to be cleared here too and in the `catch` below.
Seeing as `isConfirmModalVisible` and `forecastIdToDelete` are always set together, I don't think we need both of them.
Whether or not the modal is show could be based solely on whether `forecastIdToDelete` is defined.